concepts/sepsis/sepsis3.sql

```diff
@@ -0,0 +1,57 @@
+WITH s1 as 
+(
+  SELECT 
+    sofa.* 
+    , soi.subject_id
+    , soi.ab_id
+    , soi.antibiotic
+    , soi.antibiotic_time
+    , soi.culture_time
+    , soi.suspected_infection
+    , soi.suspected_infection_time
+    , soi.specimen
+    , soi.positive_culture
+  FROM `lcp-internal.sepsis_test.suspicion_of_infection` as soi
+  INNER JOIN `lcp-internal.sepsis_test.sofa` as sofa
+    ON soi.stay_id = sofa.stay_id 
+    AND sofa.starttime >= DATETIME_SUB(soi.suspected_infection_time, INTERVAL 48 HOUR)
+    AND sofa.endtime <= DATETIME_ADD(soi.suspected_infection_time, INTERVAL 24 HOUR)
+  WHERE sofa.stay_id is not null and soi.stay_id is not null
+)
+, s2 as 
+(
+  SELECT distinct 
+    stay_id, subject_id
+    , suspected_infection
+    , suspected_infection_time
+    , starttime, endtime
+    , respiration
+    , coagulation
+    , liver
+    , cardiovascular
+    , cns
+    , renal
+    , coalesce(respiration, 0)
+      + coalesce(coagulation, 0)
+      + coalesce(liver, 0)
+      + coalesce(cardiovascular, 0)
+      + coalesce(cns, 0)
+      + coalesce(renal, 0) as sofa_score
```

Comment:
these columns are the "instantaneous" score, so this will only be >= 2 if they have had measurements at this exact time which set them over 2. So they could have a blood gas an hour before this which indicates respiratory failure, but their score for *this* row will be 0. I'd swap this for the 24 hour columns, which is closer to the original intention of SOFA (a daily assessment).

Reply:
In that case, should line 17 become `sofa.starttime >= DATETIME_SUB(soi.suspected_infection_time, INTERVAL 24 HOUR)` since the sofa24 hour column already goes back 24 hours? If we left it as 48 I think we would technically be looking over a 72-hour range.

Reply:
Mmm I don't think so. SOFA would look back 24 hours, and you'd get a value at time T which corresponds to [T - 24, T]. Then you'd compare that to any suspected infection between [T - 48, T + 24], so it would still only look back a max of 48 hours.